components/bundle/gradle/dependencies.gradle

```diff
@@ -0,0 +1,10 @@
+dependencies {
+    implementation 'jakarta.annotation:jakarta.annotation-api:2.1.1'
+
+    api project(':components:abstractions')
```

Comment:
I initially was thinking the same, however there are some APIs like the serialization helpers that exist in the serialization module.
https://github.com/microsoft/kiota-java/blob/c1a26bf8e247f97f5d3c32d797d2a99c602040ed/components/abstractions/src/main/java/com/microsoft/kiota/serialization/KiotaJsonSerialization.java#L12

So, I think being able to access them would be useful given that from the perspective of the consumer pulling this package only, they may want access to the other APIs there...

Reply:
do you mean for the scenario where they want to register the providers manually?

Reply:
Sorry. I misread something and inferred the serialization helpers were in the serialization packages (they are actually in abstractions). So, it would block calling the code below. 

```java
KiotaJsonSerialization.serializeAsString(newUser);
```

Resolving this for now and changing the serialization packages to use `implementation`